components/atoms/PhoneNumberInput/CountrySelector/CountrySelectorIndex.vue

```diff
@@ -0,0 +1,264 @@
+                class="country-selector-list"
+            >
+                <RecycleScroller
+                    v-slot="{ item }"
+                    :item-size="1"
+                    :items="countriesSorted"
+                    key-field="iso2"
+                >
+                    <button
+                        :key="`item-${item.code}`"
+                        :class="[
+                            { selected: value === item.iso2 },
+                            { 'keyboard-selected': value !== item.iso2 && tmpValue === item.iso2 },
+                        ]"
+                        :style="[
+                            itemHeight,
+                            value === item.iso2 ? { backgroundColor: '#6d1c5d' } : {},
+                        ]"
+                        class="flex align-center country-selector-list-item"
+                        tabindex="-1"
+                        type="button"
+                        @click.stop="updateValue(item.iso2)"
+                    >
+                        <div class="country-selector-list-item-flag-container">
+                            <div :class="`iti-flag-small iti-flag ${item.iso2.toLowerCase()}`" />
+                        </div>
+                        <div class="dots-text">
+                            {{ item.name }}
+                        </div>
+                    </button>
+                </RecycleScroller>
+            </div>
+        </Transition>
+    </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, defineProps, ref, watch } from 'vue'
+import { getCountryCallingCode } from 'libphonenumber-js'
+import { RecycleScroller } from 'vue-virtual-scroller'
+import { nextTick } from '@nuxtjs/composition-api'
+
+const emit = defineEmits(['input', 'open', 'close'])
+
+const props = withDefaults(
+    defineProps<{
+        id: string
+        value: string
+        label: string
+        hint: string
+        valid: boolean
+        items: string[]
+        onlyCountries: string[]
+        ignoredCountries: string[]
+        countriesHeight: number
+    }>(),
+    {
+        id: 'CountrySelector',
+        value: '',
+        label: 'Choose country',
+        hint: '',
+        valid: false,
+        items: () => [''],
+        onlyCountries: () => [''],
+        ignoredCountries: () => [''],
+        countriesHeight: 35,
+    },
+)
+
+const isFocus = ref(false)
+const hasListOpen = ref(false)
+const tmpValue = ref<string | null>(props.value)
+const query = ref('')
+const isHover = ref(false)
+const countriesListRef = ref<HTMLElement | null>(null)
+const CountrySelectorRef = ref<HTMLElement | null>(null)
+let queryTimer: number | undefined
+
+const itemHeight = computed(() => {
+    return { height: `${props.countriesHeight}px` }
+})
+const listHeight = computed(() => {
+    return {
+        height: `${(props.countriesHeight + 1) * 7}px`,
+        maxHeight: `${(props.countriesHeight + 1) * 7}px`,
```

Comment:
Any reason why we are calculating maxHeight as well? Shouldn't height suffice?

Reply:
No, I removed it as there was no use